ResearchKit/Common/ORKImageCaptureStepViewController.m

```diff
@@ -0,0 +1,276 @@
+@property (nonatomic, strong) AVCaptureSession *captureSession;
+@property (nonatomic, strong) AVCaptureStillImageOutput *stillImageOutput;
+@property (nonatomic, strong) NSData *capturedImageData;
+@property (nonatomic, strong) NSURL *fileUrl;
+
+@end
+
+
+@implementation ORKImageCaptureStepViewController
+
+- (instancetype)initWithStep:(ORKStep *)step result:(ORKResult *)result {
+    self = [self initWithStep:step];
+    if (self) {
+        ORKStepResult *stepResult = (ORKStepResult *)result;
+        if (stepResult && [stepResult results].count > 0) {
+            ORKFileResult *fileResult = [[stepResult results] firstObject];
+            if(fileResult.fileURL) {
+                self.fileUrl = fileResult.fileURL;
+                //this will actually cause the fileURL above to be removed..
+                self.capturedImageData = [NSData dataWithContentsOfURL:fileResult.fileURL];
+            }
+        }
+    }
+    return self;
+}
+
+- (instancetype)initWithStep:(ORKStep *)step {
+    self = [super initWithStep:step];
+    if (self) {
+        _imageCaptureView = [[ORKImageCaptureView alloc] initWithFrame:CGRectZero];
+        _imageCaptureView.imageCaptureStep = (ORKImageCaptureStep *)step;
+        _imageCaptureView.delegate = self;
+        [self.view addSubview:_imageCaptureView];
+        
+        NSDictionary *dictionary = NSDictionaryOfVariableBindings(_imageCaptureView);
+        ORKEnableAutoLayoutForViews([dictionary allValues]);
+        [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_imageCaptureView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+        [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_imageCaptureView]|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:dictionary]];
+    }
+    return self;
+}
+
+- (void)setContinueButtonItem:(UIBarButtonItem *)continueButtonItem {
+    [super setContinueButtonItem:continueButtonItem];
+    _imageCaptureView.continueButtonItem = continueButtonItem;
+}
+
+- (void)retakePressed:(void (^ __nullable)())handler {
+    // Start the capture session, and reset the captured image to nil
+    dispatch_async(self.sessionQueue, ^{
+        [self.captureSession startRunning];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            self.capturedImageData = nil;
+            if(handler) {
+                handler();
+            }
+        });
+    });
+}
+
+- (void)capturePressed:(void (^ __nullable)(BOOL))handler {
+    // Capture the image via the output
+    dispatch_async(self.sessionQueue, ^{
+    	[[self stillImageOutput] captureStillImageAsynchronouslyFromConnection:[[self stillImageOutput] connectionWithMediaType:AVMediaTypeVideo] completionHandler:^(CMSampleBufferRef imageDataSampleBuffer, NSError *error) {
+            [self queue_captureImageFromData:imageDataSampleBuffer withHandler:handler];
+    	}];
+    });
+}
```

Comment:
Excellent point.  Regardless of whether it's okay if this is called multiple times in quick succession, it was not what I was intending to be possible, and so I've modified the code to prevent it from occurring.  I made the change in the `ORKImageCaptureView`, however, as that is where the button actions are targeted.  I took the same approach as used in `ORKNavigationContainerView` where a BOOL is used in a similar situation.